### PR TITLE
Viewmenu

### PR DESCRIPTION
**Pull request by michaelphagen**
Adds Viewmenu to calendar bar, comments out morevertmenu

### CODE REVIEW

File: src/components/ViewMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  CalendarView,
+  CalendarUIProps,
+} from "../calendar/types";
+
+const options = [
+  ["Schedule", "listWeek"],
+  ["Day View", "resourceTimeGridDay"],
+  ["Week View", "resourceTimeGridWeek"],
+  ["Month View", "dayGridMonth"],
+];
```
Comment (dwmorrin) [inline src/components/ViewMenu.tsx:20]: rewrite these as
```
const options = [
  {
    label: "Schedule",
    view: "listWeek",
    icon: ViewAgendaIcon,
  },
  {
    label: "Day View",
    view: "resourceTimeGridDay",
    icon: ViewDayIcon,
  },
  {
    label: "Week View",
    view: "resourceTimeGridWeek",
    icon: ViewWeekIcon,
  },
  {
    label: "Month View",
    view: "dayGridMonth",
    icon: ViewComfyIcon,
  },
];
```

File: src/components/ViewMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import Button from "@material-ui/core/Button";
+import Menu from "@material-ui/core/Menu";
+import MenuItem from "@material-ui/core/MenuItem";
+import ViewWeekIcon from "@material-ui/icons/ViewWeek";
+import ViewDayIcon from "@material-ui/icons/ViewDay";
+import ViewComfyIcon from "@material-ui/icons/ViewComfy";
+import ViewAgendaIcon from "@material-ui/icons/ViewAgenda";
+import {
+  CalendarAction,
+  CalendarView,
```
Comment (dwmorrin) [inline src/components/ViewMenu.tsx:11]: this should be deletable after proposed changes

File: src/components/ViewMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                case "Day View":
+                  return <ViewDayIcon />;
+                case "Week View":
+                  return <ViewWeekIcon />;
+                case "Schedule View":
+                  return <ViewAgendaIcon />;
+                default:
+                  return <ViewComfyIcon />;
+              }
+            })()}
```
Comment (dwmorrin) [inline src/components/ViewMenu.tsx:82]: `{icon}`

File: src/components/ViewMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        open={open}
+        onClose={handleClose}
+        PaperProps={{
+          style: {
+            maxHeight: ITEM_HEIGHT * 4.5,
+            width: "20ch",
+          },
+        }}
+      >
+        {options.map((option) => (
```
Comment (dwmorrin) [inline src/components/ViewMenu.tsx:61]: `{options.map(({label, view, icon}) => (`

File: src/components/ViewMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+          <MenuItem
+            key={option[0]}
```
Comment (dwmorrin) [inline src/components/ViewMenu.tsx:63]: `key={label}`

File: src/components/ViewMenu.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            onClick={(): void =>
+              dispatch({
+                type: CalendarAction.ChangedView,
+                payload: { currentView: option[1] as CalendarView },
```
Comment (dwmorrin) [inline src/components/ViewMenu.tsx:67]: `payload: { currentView: view },` (assertion is _probably_ not needed anymore)

File: src/components/ViewMenu.tsx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+                  return <ViewDayIcon />;
+                case "Week View":
+                  return <ViewWeekIcon />;
+                case "Schedule View":
+                  return <ViewAgendaIcon />;
+                default:
+                  return <ViewComfyIcon />;
+              }
+            })()}
+            <p style={{ paddingLeft: "15px" }}>{option[0]}</p>
```
Comment (dwmorrin) [inline src/components/ViewMenu.tsx:83]: `<p style={{ paddingLeft: "15px" }}>{label}</p>`

File: src/components/ViewMenu.tsx (hunk 2 above)
Comment (michaelphagen) [reply src/components/ViewMenu.tsx:11]: You mean CalendarView can be deleted, right? the others are needed as configured, at least how i'm looking at it.

Comment (dwmorrin) [reply src/components/ViewMenu.tsx:11]: Yes, that was meant to just be on line # 11

**Pull request merged**